sw/ground_segment/joystick/input2ivy.ml

```diff
@@ -219,19 +219,30 @@ let parse_input = fun input ->
     (name, value))
     (Xml.children input)
 
-(** Parse a '� la C' expression *)
-let parse_value = fun s ->
-  Fp_proc.parse_expression s
+(** Parse a '� la C' expression *)
+let parse_value s =
+  (* Printf.printf "s: %s\n" s; *)
+  let substrings = String.split_on_char ';' s in
+  let parsed_exprs = List.map (fun sub ->
+    let trimmed_sub = String.trim sub in
+    let parsed_expr = Fp_proc.parse_expression trimmed_sub in
+    (* Printf.printf "parsed_expr: %s\n" (Syntax.sprint parsed_expr); *)
+    parsed_expr
+  ) substrings in
+  parsed_exprs
 
 (** Parse a message field and eval *)
 let parse_msg_field = fun msg_descr field ->
   let name = Xml.attrib field "name" in
   let field_descr = try List.assoc name msg_descr.PprzLink.fields with _ ->
     Printf.printf "parse_msg_field: field %s not found\n" name;
     raise (Failure "field not found") in
-
-  let value = eval_settings_and_blocks field_descr (parse_value (Xml.attrib field "value")) in
-  (name, field_descr.PprzLink._type, value)
+  (* Printf.printf "check 2\n"; *)
+  let value = List.map (eval_settings_and_blocks field_descr) (parse_value (Xml.attrib field "value")) in
+  let field_type = field_descr.PprzLink._type in
+  let field_type_str = PprzLink.string_of_type field_type in
+  (* Printf.printf "Field name: %s, Type: %s\n" name field_type_str; *)
```

Comment:
`field_type_str` can be removed (and commented line below as well), same thing below in `execute_action` function.

Reply:
Done